core/store/ledgerstore/tx_handler.go

```diff
@@ -283,12 +286,16 @@ func handleShardCommitMsg(msg *xshard_types.XShardCommitMsg, store store.LedgerS
 	for _, addr := range txState.LockedAddress {
 		delete(lockedAddress, addr)
 	}
+	// unlock key
+	for _, key := range txState.LockedKeys {
+		unlockKey(lockedKeys, key)
```

Comment:
you need return the modifyed value from `unlockKey`, or else the length of the original `lockedKeys` will keep unchanged.